src/app/modules/form/admin-form/__tests__/admin-form.controller.spec.ts

```diff
@@ -7302,6 +7302,220 @@ describe('admin-form.controller', () => {
     })
   })
 
+  describe('handleDuplicateFormField', () => {
+    const MOCK_USER_ID = new ObjectId().toHexString()
+    const MOCK_FORM_ID = new ObjectId().toHexString()
+
+    const MOCK_USER = {
+      _id: MOCK_USER_ID,
+      email: 'somerandom@example.com',
+    } as IPopulatedUser
+    const MOCK_FIELDS = [generateDefaultField(BasicField.Rating)]
+    const MOCK_FIELD_ID = String(MOCK_FIELDS[0]._id)
+
+    const MOCK_DUPLICATED_FIELD = generateDefaultField(BasicField.Rating)
+
+    const MOCK_FORM = {
+      admin: MOCK_USER,
+      _id: MOCK_FORM_ID,
+      form_fields: MOCK_FIELDS,
+      title: 'mock title',
+    } as IPopulatedForm
+
+    const MOCK_REQ = expressHandler.mockRequest({
+      params: {
+        formId: MOCK_FORM_ID,
+        fieldId: MOCK_FIELD_ID,
+      },
+      session: {
+        user: {
+          _id: MOCK_USER_ID,
+        },
+      },
+    })
+
+    beforeEach(() => {
+      // Mock various services to return expected results.
+      MockUserService.getPopulatedUserById.mockReturnValue(okAsync(MOCK_USER))
+      MockAuthService.getFormAfterPermissionChecks.mockReturnValue(
+        okAsync(MOCK_FORM),
+      )
+      MockAdminFormService.duplicateFormField.mockReturnValue(
+        okAsync(MOCK_DUPLICATED_FIELD),
+      )
+    })
+
+    it('should return 200 when duplication is successful', async () => {
+      // Arrange
+      const mockRes = expressHandler.mockResponse()
+
+      // Act
+      await AdminFormController.handleDuplicateFormField(
+        MOCK_REQ,
+        mockRes,
+        jest.fn(),
+      )
+
+      // Assert
+      expect(mockRes.status).toHaveBeenCalledWith(200)
+      expect(mockRes.json).toHaveBeenCalledWith(MOCK_DUPLICATED_FIELD)
+      expect(MockAdminFormService.duplicateFormField).toHaveBeenCalledWith(
```

Comment:
for completeness, you can also check that the other services (`UserService`, `AuthService`) were called correctly

Reply:
or not called